WordPress/gutenberg
# Components: Remove unexpected `has-text` class when empty children are passed to `Button`

Fix the problem found in #43802's [comment](https://github.com/WordPress/gutenberg/pull/43802#discussion_r971849220).

## What?
This PR fixes unexpected `has-text` class being given to a `Button` component when an empty child is passed to it.

## Why?
As I have investigated in [this comment](https://github.com/WordPress/gutenberg/pull/43802#discussion_r971914851), when a `Button` component is placed directly under a `Tooltip` component, the Button component has an array with null as a child and is given the `has-text` class, which is not expected.
Similarly, I found that not specifying a child, such as `<Button></Button>`, or passing an empty fragment caused the same problem.
I think that this problem is caused by the laxity of the conditions for checking whether the `Button` has children or not.
## How?
The conditions for granting the has-text class have been made stricter.

## Testing Instructions
- Use the following code to override `edit.js` in any of the blocks:

<details>
<summary>Code</summary>

```javascript
/**
 * WordPress dependencies
 */
import { useBlockProps } from '@wordpress/block-editor';
import { Button, Tooltip } from '@wordpress/components';
import { help } from '@wordpress/icons';

export default function CodeEdit() {
	const blockProps = useBlockProps();
	return (
		<div { ...blockProps }>
			<p>Button with no children</p>
			<Button icon={ help } label="label" variant="primary"></Button>

			<p>Button with empty flagment children</p>
			<Button icon={ help } label="label" variant="primary">
				<></>
			</Button>

			<p>Button with Tooltip</p>
			<Tooltip text="Tooltip!">
				<Button icon={ help } label="label" variant="primary" />
			</Tooltip>

			<p>Should have .has-text class as in the past</p>
			<Button icon={ help } label="label" variant="primary">
				Push Me
			</Button>
		</div>
	);
}
```
</details>

- Insert this block and confirm that there is no has-text class  in the buttons passed empty children (this means that the button will be displayed as a square).
- Confirm that the tooltips for the buttons are also displayed as before.

## Screenshot and Screencast

| on trunk | on this branch |
| --- | --- |
| ![trunk](https://user-images.githubusercontent.com/54422211/190434578-1a3d89ab-1295-4986-978d-65685a25faff.png) | ![pr](https://user-images.githubusercontent.com/54422211/190434591-2d865284-0d33-4cec-83b2-e34e1221832f.png) |
